generative/inferers/inferer.py

```diff
@@ -303,11 +304,26 @@ class LatentDiffusionInferer(DiffusionInferer):
         scheduler: a scheduler to be used in combination with `unet` to denoise the encoded image latents.
         scale_factor: scale factor to multiply the values of the latent representation before processing it by the
             second stage.
+        ldm_latent_shape: desired SPATIAL latent space shape. Used if there is a difference in output VAE latent shape
+        and LDM shape.
+        vae_latent_shape: VAE SPATIAL latent space shape. Used if there is a difference in output VAE latent shape and
+        LDM shape.
     """
 
-    def __init__(self, scheduler: nn.Module, scale_factor: float = 1.0) -> None:
+    def __init__(self, scheduler: nn.Module, scale_factor: float = 1.0,
+                 ldm_latent_shape: list | None = None,
+                 vae_latent_shape: list | None = None) -> None:
+
         super().__init__(scheduler=scheduler)
         self.scale_factor = scale_factor
+        if (ldm_latent_shape is None) ^ (vae_latent_shape is None):
+            raise ValueError("If ldm_latent_shape is None, vae_latent_shape must be None"
+                             "and vice versa.")
+        self.ldm_latent_shape = ldm_latent_shape
+        self.vae_latent_shape = vae_latent_shape
+        if self.ldm_latent_shape is not None:
+            self.padder = SpatialPad(spatial_size=[-1,]+self.ldm_latent_shape)
```

Comment:
i can imagine cases where the user wants to crop the latent-space, rather than pad (for example, imagine if the choice is between just cropping off a single pixel border or padding a large amount which will cost lots of memory). as far as i can tell from playing with the tests your code works in this situation, but perhaps we should make these names more general. something like `autoencoder_resizer` and `ldm_resizer`, perhaps?